lib/pedicel/base.rb

```diff
@@ -142,15 +136,20 @@ def verify_signature(ca_certificate_pem: Pedicel.config[:apple_root_ca_g3_cert_p
 
     private
 
-    def self.verify_signature_certificate_oids(signature:)
-      leaf = signature.certificates.find{|c| c.extensions.find{|e| e.oid == Pedicel.config[:oids][:leaf_certificate]}}
+    def self.verify_signature_certificate_oids(signature:, config: Pedicel.config)
+      leaf = signature.certificates.find do |c|
+        c.extensions.find do |e|
+          e.oid == config[:oids][:leaf_certificate]
+        end
+      end
+
       unless leaf
-        raise SignatureError, "no leaf certificate found (OID #{Pedicel.config[:oids][:leaf_certificate]})"
+        raise SignatureError, "no leaf certificate found (OID #{config[:oids][:leaf_certificate]})"
       end
 
-      intermediate = signature.certificates.find{|c| c.extensions.find{|e| e.oid == Pedicel.config[:oids][:intermediate_certificate]}}
+      intermediate = signature.certificates.find{|c| c.extensions.find{|e| e.oid == config[:oids][:intermediate_certificate]}}
       unless intermediate
-        raise SignatureError, "no intermediate certificate found (OID #{Pedicel.config[:oids][:leaf_certificate]})"
+        raise SignatureError, "no intermediate certificate found (OID #{config[:oids][:leaf_certificate]})"
```

Comment:
I subjectively like the class/module name to specify the scope mostly because it is much easier to understand where to find the method IMO; thus, concerning "readability".

Reply:
Yeah, ruby is funny like that.
Hmm. There's also the question that a class function fetches an instance variable:
```ruby
  def self.config
    @config ||= DEFAULTS
  end
```
I'm just going to think on this for a bit.

Reply:
In this case `config` is an argument to the function.